src/explore-education-statistics-common/src/components/form/FormComboBox.tsx

```diff
@@ -156,158 +158,210 @@ const FormComboBox = ({
+            aria-labelledby={listBoxLabelId}
+            className={classNames(styles.options, {
+              [styles.optionsNoFocus]: renderedOptions.length === 0,
+            })}
+            id={`${id}-options`}
+            ref={listBoxRef}
+            role="listbox"
+            tabIndex={-1}
+            onKeyDown={event => {
+              selectNextOption(event);
+
+              const inputEl = inputRef.current;
 
-                    default:
+              if (inputEl) {
+                switch (event.key) {
+                  case 'ArrowLeft':
+                  case 'ArrowRight': {
+                    const directionChange = event.key === 'ArrowLeft' ? -1 : 1;
+
+                    inputEl.selectionStart = inputEl.selectionStart
+                      ? inputEl.selectionStart + directionChange
+                      : 0;
+                    inputEl.selectionEnd = inputEl.selectionStart;
+                    inputEl.focus();
+                    break;
                   }
+
+                  case 'Enter':
+                    onSelect(selectedOption);
+                    toggleShowOptions(false);
+                    break;
+
+                  case 'Home':
+                    inputEl.selectionStart = 0;
+                    inputEl.selectionEnd = 0;
+                    inputEl.focus();
+                    break;
+
+                  case 'End':
+                    inputEl.selectionStart = inputEl.value.length;
+                    inputEl.selectionEnd = inputEl.selectionStart;
+                    inputEl.focus();
+                    break;
+
+                  case 'Escape':
+                    setValue('');
+                    toggleShowOptions(false);
+                    inputEl.focus();
+                    break;
+
+                  default:
                 }
-              }}
-            >
-              {renderedOptions &&
-                renderedOptions.length > 0 &&
-                renderedOptions.map((item, index) => {
-                  const key = index;
-
-                  return (
-                    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
-                    <li
-                      aria-selected={selectedOption === index}
-                      key={key}
-                      id={`${id}-option-${index}`}
-                      className={
-                        selectedOption === index ? styles.selected : undefined
-                      }
-                      role="option"
-                      ref={el => {
-                        if (el) {
+              }
+            }}
+          >
+            {renderedOptions &&
+              renderedOptions.length > 0 &&
+              renderedOptions.map((item, index) => {
+                const key = index;
+
+                return (
+                  // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+                  <li
+                    aria-selected={selectedOption === index}
```

Comment:
We could refactor all of the `selectedOption === index` expressions into a single `isSelected` variable and re-use this across the various props